### PR TITLE
fix: allow val_check_interval to be a float in (0, 1] or an int >= 1

### LINKED ISSUE #627: allow val_check_interval to be a float
In pytorch, val_check_interval can be either a float in the range [0,1] or an int.  If it's a float, it specifies how frequently the validation set is analyzed per epoch, whereas if it's an int it specifies the same thing in number of batches (steps).  Currently, casanovo only allows it to be an int. I think we should allow either, and document it as such in the config file.  Furthermore, I think the default value should be 1.0.

### PR DESCRIPTION
**Pull request by Ghraven**
Closes #627

## What

Adds `_parse_val_check_interval()` — a dedicated validator that accepts PyTorch Lightning's full range for `val_check_interval`:

- **`float` in `(0.0, 1.0]`** — fraction of an epoch between validations
- **`int >= 1`** — number of training batches between validations

Previously the `_config_types` dict mapped `val_check_interval` to `int`, so passing `0.25` or `1.0` would be silently coerced or raise a confusing error.

## Changes

- `casanovo/config.py`: replace `val_check_interval=int` with `val_check_interval=_parse_val_check_interval`
- New module-level function `_parse_val_check_interval` with clear docstring and `ValueError` for out-of-range values
- Default value in `config.yaml` (`50_000`) continues to work as an int

## Example

```yaml
# validate every quarter of an epoch
val_check_interval: 0.25

# validate every 10 000 steps (existing behaviour)
val_check_interval: 10000
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced validation of the validation check interval configuration parameter to properly handle different input types and provide clearer error messages when invalid values are supplied.

### CODE REVIEW

**Comment by Lilferrit** [comment]
I would hold off on implementing this until #545 is implemented (for which there is a PR open already, #590). The implementation of this will change once we switch over to omgaconf for handling the config parsing.

**Comment by Ghraven** [comment]
Thanks for the context, @Lilferrit — I was not aware of #590 or the planned omegaconf migration in #545. I will hold off on this until that work lands. Happy to revisit and rebase on top of the new config structure once #590 is merged if that would still be useful.

**Comment by Ghraven** [comment]
Hi! Just a gentle check-in on this one. If you'd like any changes to the validation logic or the approach isn't quite right for the project, I'm happy to revise. No pressure at all — thanks so much for maintaining this!

**Comment by bittremieux** [comment]
As mentioned before, please follow our [contributing guidelines](https://github.com/Noble-Lab/casanovo/blob/main/CONTRIBUTING.md). One important aspect that's mentioned there is that new contributions should be submitted against `dev`, never against `main`. I'll be closing all PRs that don't adhere to these guidelines.
